streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala

```diff
@@ -454,6 +454,39 @@ object ZStream extends Serializable {
       } yield Control(pull, Command.noop)
     }
 
+  /**
+   * Creates a stream from an iterable collection of values
+   *
+   * @tparam A the iterable value type
+   * @param as a iterable collection of values
+   * @return a stream which emits values from iterable collection
+   */
+  def fromIterable[A](as: => Iterable[A]): ZStream[Any, Nothing, Any, Unit, A] =
+    fromChunk(Chunk.fromIterable(as))
+
+  /**
+   * Creates a stream from an effect that produces iterator
+   *
+   * @tparam R the environment type
+   * @tparam E the error type
+   * @tparam A the value type of elements produced by iterator
+   * @param iterator effect that produces iterator
+   * @return a stream which emits values produced by iterator
+   */
+  def fromIterator[R, E, A](iterator: ZIO[R, E, Iterator[A]]): ZStream[R, E, Any, Unit, A] =
+    fromEffect(iterator).flatMap(iter =>
+      ZStream {
+        Managed.effectTotal {
+          Control(
+            IO.effectSuspendTotal {
+              if (iter.hasNext) UIO.succeed(iter.next()) else Pull.endUnit
```

Comment:
```suggestion
              if (iter.hasNext) UIO.succeedNow(iter.next()) else Pull.endUnit
```
☝️ I think this one _must_ be strict